Develop

TsvetanKichuk opened the pull request.

TsvetanKichuk committed:
Add quote scraping and CSV export functionality

TsvetanKichuk committed:
Refactor parser and add author bio caching with CSV output

Refactored the parsing logic to simplify code structure and improve readability. Added functionality to retrieve and cache author bios for inclusion in the output. Updated the main function to support saving both quotes and author bios into separate CSV files.

TsvetanKichuk committed:
Refactor parsing functions by removing redundant comments.

Removed unnecessary comments that restated function or variable names directly, improving code readability. This makes the code cleaner and focuses on self-explanatory names and functionality.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **Something went wrong while processing the review.**
> <sub>*[CODE: 2]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in 1 hour.
If you believe this is an error or need assistance, please contact the support team.

TsvetanKichuk committed:
Refactor and add type hints to parsing and CSV functions

Reorganized imports, introduced type hints for better readability, and ensured consistent formatting throughout the parsing script. This enhances code maintainability and enforces stricter input/output expectations.

TsvetanKichuk committed:
Refactor quote parsing and fix tags formatting

Introduced a Quote dataclass to improve structure in `parse.py`. Adjusted CSV output to ensure consistent formatting, especially for tags, by converting them into list representations.

TsvetanKichuk committed:
Remove unused 'author_url' attribute from Quote class

mateacademy-ai-mentor commented on app/parse.py:
```diff
@@ -1,16 +1,88 @@
+    tags: List[str]
+
+
+def get_quotes_from_page(url: str) -> (
+        Tuple)[List[Dict[str, str]], Optional[str]]:
```
There is a syntax error in the return type annotation of the `get_quotes_from_page` function. The correct syntax should be `Tuple[List[Dict[str, str]], Optional[str]]` instead of `(Tuple)[List[Dict[str, str]], Optional[str]]`.

mateacademy-ai-mentor commented on app/parse.py:
```diff
@@ -1,16 +1,88 @@
 
 if __name__ == "__main__":
-    main("quotes.csv")
+    main("output_quotes_csv")
```
The `main` function is called with only one argument, `output_quotes_csv`. If you intend to use the `output_authors_csv` functionality, you should provide a second argument for it. If not, consider removing the optional parameter from the function definition.

mateacademy-ai-mentor reviewed (approved):
Great job on implementing the web scraping functionality! 🎉 The code meets the task requirements and successfully scrapes quotes from the website, handling pagination effectively. There are a couple of minor issues to address: the syntax error in the return type annotation of `get_quotes_from_page` and the optional `output_authors_csv` parameter in the `main` function. These don't impact the core functionality, so feel free to address them in future improvements. Keep up the good work! 🚀

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨
